nck/readers/googleads_reader.py

```diff
@@ -246,19 +249,30 @@ def get_report_definition(self):
     def add_period_to_report_definition(self, report_definition):
         """Add Date period from provided start date and end date, when CUSTOM DATE range is called"""
         if (self.date_range_type == "CUSTOM_DATE") & (not self.start_date or not self.end_date):
-            logger.warning(
-                "Custom Date Range selected but no date range provided :" + DATE_RANGE_TYPE_POSSIBLE_VALUES[0] + " by default"
+            raise NoDateDefinitionException(
+                "You must define a couple \
+            start-date/end-date when using a custom_date"
             )
-            logger.warning("https://developers.google.com/adwords/api/docs/guides/reporting#custom_date_ranges")
-            report_definition["dateRangeType"] = DATE_RANGE_TYPE_POSSIBLE_VALUES[0]
         elif self.date_range_type == "CUSTOM_DATE":
-            logger.info("Date format used for request : Custom Date Range with start_date and end_date provided")
+            logger.info(
+                "Date format used for request : Custom Date Range with\
+            start_date and end_date provided"
+            )
             report_definition["selector"]["dateRange"] = self.create_date_range(self.start_date, self.end_date)
+        elif self.start_date is not None and self.end_date is not None and self.date_range_type != "CUSTOM_DATE":
+            raise InconsistentDateDefinitionException(
+                "You must define either the couple start_date and end_date \
+            or a date_range \(different from CUSTOM_DATE\), but not both"
+            )
```

Comment:
There is a function named `check_date_range_definition_conformity` that seems to do the same thing. Why not use it here?

Reply:
I cannot use this function and that's because of the way google_ads has been coded initially. The parameter date_range_type is always defined (it has a default parameter in the click sequence, and I decided not to touch it).

Reply:
Ok. I think we should remove the default parameters from all readers as it goes against our will that is to remove all implicit behaviours. So, feel free to remove them and use your function. It is here for that exact purpose.